pennylane/transforms/classical_jacobian.py

```diff
@@ -19,8 +19,55 @@
 
 
 def classical_jacobian(qnode):
-    """Function to extract the Jacobian
-    matrix of the classical part of a QNode"""
+    r"""Returns a function to extract the Jacobian
+    matrix of the classical part of a QNode.
+
+    This transform allows the classical dependence between the QNode
+    arguments and the quantum gate arguments to be extracted.
+
+    Args:
+        qnode (.QNode): QNode to compute the (classical) Jacobian of
+
+    Returns:
+        function: Function which accepts the same arguments as the QNode.
+        When called, this function will return the Jacobian of the QNode
+        gate arguments with respect to the QNode arguments.
+
+    **Example**
+
+    Consider the following QNode:
+
+    >>> @qml.qnode(dev)
+    ... def circuit(weights):
+    ...     qml.RX(weights[0], wires=0)
+    ...     qml.RY(weights[0], wires=1)
+    ...     qml.RZ(weights[2] ** 2, wires=1)
+    ...     return qml.expval(qml.PauliZ(0))
+
+    We can use this transform to extract the relationship :math:`f: \mathbb{R}^n \rightarrow
+    \mathbb{R}^m` between the input QNode arguments :math:`w` and the gate arguments :math:`g`, for
+    a given value of the QNode arguments:
+
+    >>> cjac_fn = qml.transforms.classical_jacobian(circuit)
+    >>> weights = np.array([1., 1., 1.], requires_grad=True)
+    >>> cjac = cjac_fn(weights)
+    >>> print(cjac)
+    [[1. 0. 0.]
+     [1. 0. 0.]
+     [0. 0. 2.]]

```

Comment:
What happens if `weight` is a higher dimensional tensor?

Reply:
Good question, let me check 🤔

Reply:
Works perfectly 😅 

```python
import pennylane as qml
from pennylane import numpy as np

dev = qml.device("default.qubit", wires=2)

@qml.qnode(dev)
def circuit(weights):
    qml.RX(weights[0, 0], wires=0)
    qml.RY(weights[0, 1], wires=1)
    qml.RZ(weights[1, 0] ** 2, wires=1)
    return qml.expval(qml.PauliZ(0))
```

where

```pycon
>>> cjac_fn = qml.transforms.classical_jacobian(circuit)
>>> weights = np.array([[1., 1.], [1., 1.]], requires_grad=True)
>>> cjac = cjac_fn(weights)
>>> print(cjac)
[[[1. 0.]
  [0. 0.]]

 [[0. 1.]
  [0. 0.]]

 [[0. 0.]
  [2. 0.]]]
```

Reply:
noice

Reply:
i love it when you don't take an edge case into consideration but it just works